src/DatePicker/DatePicker.js

```diff
@@ -309,11 +303,11 @@ class DatePicker extends React.Component {
       <div style={prepareStyles(Object.assign({}, style))}>
         <TextField
           {...other}
-          style={textFieldStyle}
-          ref="input"
-          value={this.state.date ? formatDate(this.state.date) : ''}
           onFocus={this.handleFocus}
           onTouchTap={this.handleTouchTap}
+          ref="input"
+          style={textFieldStyle}
+          value={this.state.date ? formatDate(this.state.date) : undefined}
```

Comment:
`undefined` needs to be an empty string `''` or react is going to throw uncontrolled/controlled errors when you select a date